test/e2e/static_pods.go

```diff
@@ -0,0 +1,103 @@
+	var podFile = "static-pod.yaml"
+	if err := ioutil.WriteFile(podFile, []byte(podYaml), 0644); err != nil {
+		framework.Failf("Unable to write static-pod.yaml  to disk: %v", err)
+	}
+	defer func() {
+		if err := os.Remove(podFile); err != nil {
+			framework.Logf("Unable to remove the static-pod.yaml from disk: %v", err)
+		}
+	}()
+	var dst = fmt.Sprintf("%s:/etc/kubernetes/manifests/%s", nodeName, podFile)
+	cmd := []string{"docker", "cp", podFile, dst}
+	framework.Logf("Running command %v", cmd)
+	_, err := runCommand(cmd...)
+	if err != nil {
+		framework.Failf("failed to copy pod file to node %s", nodeName)
+	}
+
+}
+
+func removeStaticPodFile(nodeName string, podFile string) {
+	cmd := []string{"docker", "exec", nodeName, "bash", "-c", "rm /etc/kubernetes/manifests/static-pod.yaml"}
+	framework.Logf("Running command %v", cmd)
+	_, err := runCommand(cmd...)
+	if err != nil {
+		framework.Failf("failed to remove pod file from node %s", nodeName)
+	}
+
+}
+
+//This test does the following
+//Applies a static-pod.yaml file to a nodes /etc/kubernetes/manifest dir
+//Expects the static pod to succeed
+var _ = ginkgo.Describe("Creating a static pod on a node", func() {
+
+	const (
+		podName      string = "static-pod"
+		podFile      string = "static-pod.yaml"
+		agnhostImage string = "k8s.gcr.io/e2e-test-images/agnhost:2.26"
+	)
+
+	f := wrappedTestFramework("staticpods")
+
+	var cs clientset.Interface
+
+	ginkgo.BeforeEach(func() {
+		cs = f.ClientSet
+	})
+
+	ginkgo.It("Should successfully create then remove a static pod", func() {
+		nodes, err := e2enode.GetBoundedReadySchedulableNodes(cs, 3)
+		framework.ExpectNoError(err)
+		if len(nodes.Items) < 1 {
+			framework.Failf("Test requires 1 Ready node, but there are none")
+		}
+		nodeName := nodes.Items[0].Name
+		podName := fmt.Sprintf("static-pod-%s", nodeName)
+		contName := "web"
+
+		ginkgo.By("copying a pod.yaml file into the /etc/kubernetes/manifests dir of a node")
+		framework.Logf("creating %s on node %s", podName, nodeName)
+		var staticPodYaml = fmt.Sprintf(`apiVersion: v1
+kind: Pod
+metadata:
+  name: static-pod
+  namespace: %s
+spec: 
+  containers: 
+    - name: web 
+      image: %s
+      command: ["/bin/bash", "-c", "trap : TERM INT; sleep infinity & wait"]
+`, f.Namespace.Name, agnhostImage)
+		createStaticPod(f, nodeName, staticPodYaml)
+		//defer removeStaticPodFile(nodeName, podFile)
+		err = e2epod.WaitForPodSuccessInNamespace(f.ClientSet, podName, f.Namespace.Name)
+		if err != nil {
+			logs, logErr := e2epod.GetPodLogs(f.ClientSet, f.Namespace.Name, podName, contName)
+			if logErr != nil {
+				framework.Logf("Warning: Failed to get logs from pod %q: %v", podName, logErr)
+			} else {
+				framework.Logf("pod %s/%s logs:\n%s", f.Namespace.Name, podName, logs)
+			}
+		}
+		ginkgo.By("Removing the pod file from the nodes /etc/kubernetes/manifests")
+		framework.Logf("Removing %s from %s", podName, nodeName)
+		removeStaticPodFile(nodeName, podFile)
```

Comment:
cool beans. And now do a getPods to make sure it is really gone ... without any bangs! :)

Reply:
had to resolve this by checking for pod success in namespace and expecting an error. Couldnt use any of the expectPodTerminated or other deletion failures from the e2e podPackage because static pod deletion does not work the same as api pod deletion. When we remove the file the pod just ceases to exist. Not quite the same as deletion. The delete functions would all error out because its like the pod never existed in the first place, but we can check and see if that podname is running in the namespace, and if it timesout (meaning it is not) then the pod must have been removed :)

Reply:
👍

Reply:
@bpickard22 why can't we use  `[WaitForPodNotFoundInNamespace](https://pkg.go.dev/k8s.io/kubernetes/test/e2e/framework/pod#WaitForPodNotFoundInNamespace)`?: 
```
WaitForPodNotFoundInNamespace returns an error if it takes too long for the pod to fully terminate. Unlike `waitForPodTerminatedInNamespace`, the pod's Phase and Reason are ignored. If the pod Get api returns IsNotFound then the wait stops and nil is returned. If the Get api returns an error other than "not found" then that error is returned and the wait stops.
```

Trying to avoid the chance of a flake if kubelet does not react immediately after the manifest is removed from disk.

Reply:
Note: going to change this to use the provided function from the framework package in a seperate pr, where i bump the versions of the deps in the go mod to 1.26